sw/device/lib/dif/dif_otbn.c

```diff
@@ -8,6 +8,25 @@
 
 #include "otbn_regs.h"  // Generated.
 
+_Static_assert(kDifOtbnErrBitBadDataAddr ==
+                   (1 << OTBN_ERR_CODE_BAD_DATA_ADDR_BIT),
+               "Layout of error bits changed.");
+_Static_assert(kDifOtbnErrBitBadInsnAddr ==
+                   (1 << OTBN_ERR_CODE_BAD_INSN_ADDR_BIT),
+               "Layout of error bits changed.");
+_Static_assert(kDifOtbnErrBitCallStack == (1 << OTBN_ERR_CODE_CALL_STACK_BIT),
+               "Layout of error bits changed.");
+_Static_assert(kDifOtbnErrBitIllegalInsn ==
+                   (1 << OTBN_ERR_CODE_ILLEGAL_INSN_BIT),
+               "Layout of error bits changed.");
+_Static_assert(kDifOtbnErrBitLoop == (1 << OTBN_ERR_CODE_LOOP_BIT),
+               "Layout of error bits changed.");
+_Static_assert(kDifOtbnErrBitFatalImem == (1 << OTBN_ERR_CODE_FATAL_IMEM_BIT),
+               "Layout of error bits changed.");
+_Static_assert(kDifOtbnErrBitFatalDmem == (1 << OTBN_ERR_CODE_FATAL_DMEM_BIT),
+               "Layout of error bits changed.");
+_Static_assert(kDifOtbnErrBitFatalReg == (1 << OTBN_ERR_CODE_FATAL_REG_BIT),
+               "Layout of error bits changed.");
```

Comment:
Instead of that static assert, can't we just use the generated bits in the definition of the kDifOtbnErrBit enum?

Reply:
I don't think we're allowed to #include the auto-generated file in the DIF header that would be needed to do this. The concept being the DIF is providing the interface and the details of the register layout are implementation specific so the header shouldn't include them.

Reply:
Ah I see, :+1: